docs/src/security.md

```diff
@@ -165,14 +171,63 @@ namespaced resources.
   validate them before starting the restore process.
 
 `nodes`
-: The operator needs to get the labels for Affinity and AntiAffinity, so it can
-  decide in which nodes a pod can be scheduled preventing the replicas to be
-  in the same node, specially if nodes are in different availability zones. This
-  permission is also used to determine if a node is schedule or not, avoiding
-  the creation of pods that cannot be created at all.
+: The operator needs to get the labels for Affinity and AntiAffinity so it can
+  decide in which nodes a pod can be scheduled. This is useful, for example, to
+  prevent the replicas from being scheduled in the same node - especially
+  important if nodes are in different availability zones. This
+  permission is also used to determine whether a node is scheduled, preventing
+  the creation of pods on unscheduled nodes, or triggering a switchover if
+  the primary lives in an unscheduled node.
 
-To see all the permissions required by the operator, you can run `kubectl
-describe clusterrole cnpg-manager`.
+
+#### Deployments and `ClusterRole` Resources
+
+As mentioned above, each deployment method may have variations in the namespace
+location of the service account, as well as the names and types of role
+bindings and respective roles.
+
+##### Via Kubernetes Manifest
+
+When installing CloudNativePG using the Kubernetes manifest, permissions are
+set to `ClusterRoleBinding` by default. You can inspect the permissions
+required by the operator by running:
+
+```sh
+kubectl describe clusterrole cnpg-manager
+```
+
+!!! Info
+    You can customize the YAML file to align with your security requirements by
+    using a `RoleBinding` instead of a `ClusterRoleBinding`. This approach enables
+    deployment of the operator in a dedicated namespace with restricted access from
+    administrators. This ensures that standard users cannot access the operator's
+    `ServiceAccount`.
```

Comment:
I don't think we explain anywhere how to do this, and it's not obvious.
Should we have this info box?
Or maybe add a section on how to customize the YAML?

Reply:
Probably we can do that on the plugin side, but that will be out of the scope of this documentation for sure

Reply:
I found this awkward too, but I understand the need to outline this possibility. I have tried several times to write a sentence that explains that CNPG doesn't support this yet, but could not come with an easy and clear way to do it. Hence, less is more and decided to omit it.

Reply:
Honestly, I was expecting the whole `cnpg generate` command to already take into account all of this, and was puzzled when I discovered it wasn't actually working. So I'd rather work on a solution than documenting the current state (which is not a feature we provide).